airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -34,11 +34,30 @@ class TriggerRuleDep(BaseTIDep):
     IGNOREABLE = True
     IS_TASK_DEP = True
 
+    @staticmethod
+    @provide_session
+    def _get_states_count_upstream_ti(ti, finished_tasks, session):
+        """
+        This function returns the states of the upstream tis for a specific ti in order to determine
+        whether this ti can run in this iteration
+
+        :param ti: the ti that we want to calculate deps for
+        :type ti: airflow.models.TaskInstance
+        :param finished_tasks: all the finished tasks of the dag_run
+        :type finished_tasks: list[airflow.models.TaskInstance]
+        """
+        if not finished_tasks:
```

Comment:
it looks like you want to check that `finished_tasks is None`,   otherwise, on every new dagrun, this query will run

Reply:
Good catch @evgenyshulman!